fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapper.kt

```diff
@@ -48,18 +53,27 @@ private fun String.toGrade(): ReviewQualityCheckState.Grade? =
         null
     }
 
-private fun Highlight.toHighlights(): Map<ReviewQualityCheckState.HighlightType, List<String>>? {
-    val highlights: Map<ReviewQualityCheckState.HighlightType, List<String>?> = mapOf(
-        ReviewQualityCheckState.HighlightType.QUALITY to quality,
-        ReviewQualityCheckState.HighlightType.PRICE to price,
-        ReviewQualityCheckState.HighlightType.SHIPPING to shipping,
-        ReviewQualityCheckState.HighlightType.PACKAGING_AND_APPEARANCE to appearance,
-        ReviewQualityCheckState.HighlightType.COMPETITIVENESS to competitiveness,
-    )
+private fun Highlight.toHighlights(): Map<HighlightType, List<String>>? =
+    HighlightType.values()
+        .associateWith { highlightsForType(it) }
+        .filterValues { it != null }
+        .mapValues { it.value!! }
+        .ifEmpty { null }
 
-    return highlights.filterValues { it != null }.mapValues { it.value!! }.ifEmpty { null }
-}
+private fun Highlight.highlightsForType(highlightType: HighlightType) =
```

Comment:
Plugging it again here 🙂 
```suggestion
private fun Highlight.highlightForType(highlightType: HighlightType) =
```
or
```suggestion
private fun Highlight.ForType(highlightType: HighlightType) =
```